cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -187,6 +207,69 @@ protected Future<Pod> pod(Integer podId) {
+                .setVerifyHost(true)
+                .setSsl(true)
+                .setVerifyHost(true)
+                .setPemTrustOptions(pto)
+                .setKeyCertOptions(pkco);
+    }
+
+    protected <T> Future<Map<String, Object>> doGet(String host) {
+        LOGGER.debugCr(reconciliation, "Making GET request to {}", BROKER_STATE_REST_PATH);
+        return HttpClientUtils.withHttpClient(vertx, getHttpClientOptions(), (httpClient, result) ->
+                httpClient.request(HttpMethod.GET, BROKER_STATE_HTTPS_PORT, host, BROKER_STATE_REST_PATH, request -> {
+                    if (request.succeeded()) {
+                        request.result().setFollowRedirects(true)
+                        .putHeader("Accept", "application/json");
+                        request.result().send(response -> {
+                            if (response.succeeded()) {
+                                if (response.result().statusCode() == 200) {
+                                    response.result().bodyHandler(body -> {
+                                        var jsonBody = body.toJsonObject().getMap();
+                                        result.complete(jsonBody);
+                                    });
+                                } else {
+                                    result.fail(new RuntimeException("Unexpected HTTP status code: " + response.result().statusCode()));
+                                }
+                            } else {
+                                result.tryFail(response.cause());
+                            }
+                        });
+                    } else {
+                        result.tryFail(request.cause());
+                    }
+                }));
+    }
+
+    private int getBrokerState(String podName) {
+        int brokerstate = -1;
+        String host = DnsNameGenerator.podDnsName(namespace, KafkaResources.brokersServiceName(cluster), podName);
+        CompletableFuture<Map<String, Object>> completableFuture = new CompletableFuture<>();
+
+        Future<Map<String, Object>> getFuture = doGet(host);
+        getFuture.onComplete(future -> {
+            if (future.succeeded()) {
+                completableFuture.complete(future.result());
+            } else {
+                completableFuture.completeExceptionally(future.cause());
+            }
+        });
+
+        try {
+            var result = completableFuture.get(3, TimeUnit.SECONDS);
+            brokerstate = (int) result.get("brokerState");
+            if (brokerstate == BROKER_RECOVERY_STATE && result.containsKey("recoveryState")) {
+                LOGGER.infoCr(reconciliation, podName + " is in log recovery. The current recovery state: " + result.get("recoveryState"));
+            }
+        } catch (TimeoutException | InterruptedException | ExecutionException e) {
+            LOGGER.warnCr(reconciliation, "Failed to get broker state", e);
+        }
+        return brokerstate;
+    }
```

Comment:
I wonder if it would make sense to factor this out into a separate class. Might make the code a bit cleaner and make it easier to mock it in tests etc.